flutter_appauth/ios/Classes/AppAuthIOSAuthorization.m

```diff
@@ -2,14 +2,19 @@
 
 @implementation AppAuthIOSAuthorization
 
-- (id<OIDExternalUserAgentSession>) performAuthorization:(OIDServiceConfiguration *)serviceConfiguration clientId:(NSString*)clientId clientSecret:(NSString*)clientSecret scopes:(NSArray *)scopes redirectUrl:(NSString*)redirectUrl additionalParameters:(NSDictionary *)additionalParameters preferEphemeralSession:(BOOL)preferEphemeralSession result:(FlutterResult)result exchangeCode:(BOOL)exchangeCode{
+- (id<OIDExternalUserAgentSession>) performAuthorization:(OIDServiceConfiguration *)serviceConfiguration clientId:(NSString*)clientId clientSecret:(NSString*)clientSecret scopes:(NSArray *)scopes redirectUrl:(NSString*)redirectUrl additionalParameters:(NSDictionary *)additionalParameters preferEphemeralSession:(BOOL)preferEphemeralSession result:(FlutterResult)result exchangeCode:(BOOL)exchangeCode nonce:(NSString*)nonce{
   OIDAuthorizationRequest *request =
   [[OIDAuthorizationRequest alloc] initWithConfiguration:serviceConfiguration
                                                 clientId:clientId
                                             clientSecret:clientSecret
-                                                  scopes:scopes
+                                                   scope:[OIDScopeUtilities scopesWithArray:scopes]
                                              redirectURL:[NSURL URLWithString:redirectUrl]
                                             responseType:OIDResponseTypeCode
+                                                   state:nil
+                                                   nonce:nonce
+                                            codeVerifier:nil
+                                           codeChallenge:nil
+                                     codeChallengeMethod:nil
```

Comment:
Setting these to nil will break backward compatibility, as the original `initWithConfiguration` sets these values.

Here's what I did for my own fork and what I suggest:
```
  NSString *codeVerifier = [OIDAuthorizationRequest generateCodeVerifier];
  NSString *codeChallenge = [OIDAuthorizationRequest codeChallengeS256ForVerifier:codeVerifier];

  OIDAuthorizationRequest *request =
  [[OIDAuthorizationRequest alloc] initWithConfiguration:serviceConfiguration
                                                clientId:clientId
                                                clientSecret:clientSecret
                                                scope:[OIDScopeUtilities scopesWithArray:scopes]
                                                redirectURL:[NSURL URLWithString:redirectUrl]
                                                responseType:OIDResponseTypeCode
                                                state:[OIDAuthorizationRequest generateState]
                                                nonce: nonce != nil ? nonce : [OIDAuthorizationRequest generateState]
                                                codeVerifier:codeVerifier
                                                codeChallenge:codeChallenge
                                                codeChallengeMethod:OIDOAuthorizationRequestCodeChallengeMethodS256
                                                additionalParameters:additionalParameters];
```

Reply:
There's another file `flutter_appauth/macos/Classes/AppAuthMacOSAuthorization.m` that you ought to change as well to ensure the same behaviour on all supported platforms. This is for Mac.